### PR TITLE
feat: apisix dashboard custom plugins configuration support

### PR DESCRIPTION
**Pull request by elliottcrif**
In more custom apisix deployments with custom plugins being used overriding schema.json and plugins in the configuration are useful features.

- feat: add support for plugins conf
- feat: add support for mounting custom schema.json
- docs: update apisix-dashboard docs

### CODE REVIEW

File: charts/apisix-dashboard/README.md [hunk 1]
```diff
@@ -73,6 +73,8 @@ _See [helm upgrade](https://helm.sh/docs/helm/helm_upgrade/) for command documen
 | config.conf.log.accessLog.filePath | string | `"/dev/stdout"` | Error log path |
 | config.conf.log.errorLog | object | `{"filePath":"/dev/stderr","level":"warn"}` | Error log level. Supports levels, lower to higher: debug, info, warn, error, panic, fatal |
 | config.conf.log.errorLog.filePath | string | `"/dev/stderr"` | Access log path |
+| config.conf.plugins | list | `[]` |  |
```
Comment (elliottcrif) [inline charts/apisix-dashboard/README.md:76]: Need to add description of the properties

Comment (lingsamuel) [reply charts/apisix-dashboard/README.md:76]: Readme file is auto-generated by `make helm-docs`. The field comment is using format like this:

```yaml
# -- comment goes here
fieldName: value
```

File: charts/apisix-dashboard/templates/configmap.yaml [hunk 2]
```diff
@@ -65,3 +65,7 @@ data:
           password: {{ .password }}
         {{- end }}
     {{- end }}
+    plugins:
+      {{- range .Values.config.conf.plugins }}
```
Comment (lingsamuel) [inline charts/apisix-dashboard/templates/configmap.yaml:69]: We should check the plugins list and generate this field only when it is non-empty

File: charts/apisix-dashboard/README.md (hunk 1 above)
Comment (elliottcrif) [reply charts/apisix-dashboard/README.md:76]: Okay awesome 🙏

File: charts/apisix-dashboard/templates/configmap.yaml (hunk 2 above)
Comment (elliottcrif) [reply charts/apisix-dashboard/templates/configmap.yaml:69]: This is a great call ✅

File: charts/apisix-dashboard/templates/configmap.yaml [hunk 3]
```diff
@@ -65,3 +65,9 @@ data:
           password: {{ .password }}
         {{- end }}
     {{- end }}
+    {{- with .Values.config.conf.plugins }}
```
Comment (elliottcrif) [inline charts/apisix-dashboard/templates/configmap.yaml:68]: @lingsamuel What do you think?

File: charts/apisix-dashboard/values.yaml [hunk 4]
```diff
@@ -71,6 +71,12 @@ securityContext: {}
   # runAsUser: 1000
 
 config:
+  schema:
+    # -- Overrides APISIX Dashboard schema.json
```
Comment (elliottcrif) [inline charts/apisix-dashboard/values.yaml:75]: Added the comments @lingsamuel

**Comment by elliottcrif** [comment]
@lingsamuel how can I request another reviewer?

**Comment by lingsamuel** [comment]
> @lingsamuel how can I request another reviewer?

CI is failing, please make it pass first

**Comment by elliottcrif** [comment]
@lingsamuel Can you trigger workflow?

**Comment by lingsamuel** [comment]
> @lingsamuel Can you trigger workflow?

Triggered. `make helm-docs` is required

**Comment by elliottcrif** [comment]
@lingsamuel ran `make helm-docs` and fixed error

**Comment by elliottcrif** [comment]
@AlinsRan Can you review 😄

**Review by AlinsRan — APPROVED** [review]

**Review by Gallardot — APPROVED** [review]

**Pull request merged**